### PR TITLE
Add passing tests to problem statement

### PR DESCRIPTION
**Pull request by huber-florian**
This PR adds passing and failing test cases to the problem statement and also changes the color of the UML font based on the tests!

Bildschirm­foto 2022-12-11 um 19 28 22

Bildschirm­foto 2022-12-11 um 19 28 51

Bildschirm­foto 2022-12-11 um 21 15 52

### CODE REVIEW

**Review by katjanakosic — APPROVED** [review]
Looks nice to me! 😁 Is the top bar showing the general overview of passed test cases also included?

**Comment by huber-florian** [comment]
> Looks nice to me! 😁 Is the top bar showing the general overview of passed test cases also included?

No not yet. Will be part of a next PR.

File: Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift [hunk 1]
```diff
@@ -32,47 +32,91 @@ class ProblemStatementCellViewModel: ObservableObject {
...
+                line = line.replacingOccurrences(of: "[task][", with: " ![Test failed](asset:///TestFailedSymbol) **")
+            } else if line.contains("[task][") && line.contains("1") {
+                line = calculateScoreOfPassingTests(line)
+                line = line.replacingOccurrences(of: "[task][", with: " ![Test passed](asset:///TestPassedSymbol) **")
+            }
+
+            problemStatementText.append(line + "\n")
+        }
+
+        return problemStatementText
```
Comment (pal03377) [inline Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift:87]: Just an idea (doesn't have to be implemented): This looks a lot like business logic, maybe it should be in a ViewModel?

File: Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift [hunk 2]
```diff
@@ -32,47 +32,91 @@ class ProblemStatementCellViewModel: ObservableObject {
...
+    private func replaceUMLTestsColor(_ problemStatement: String, _ feedbacks: [AssessmentFeedback], _ colorSchemeVariable: ColorScheme) -> String {
+        var plantUML = problemStatement
+
+        for feedback in feedbacks {
+            if let feedbackText = feedback.text {
+                plantUML = plantUML.replacingOccurrences(
+                    of: "testsColor(\(feedbackText))",
+                    with: feedback.credits == 0.0 ? "red" : "green")
+            }
+        }
```
Comment (pal03377) [inline Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift:115]: Also could be in a ViewModel?

**Review by pal03377 — APPROVED** [review]
Very well done! The icons look really fitting (I guess they are the Artemis ones? Still also fit nice into the app). At some later point we might also want to show the actual test errors somewhere, but that will be a feature for later I think.

**Comment by botrufus** [comment]
> At some later point we might also want to show the actual test errors somewhere, but that will be a feature for later I think.

Those are already shown in the feedback tab. @pal03377

File: Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift (hunk 1 above)
Comment (huber-florian) [reply Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift:87]: It is in a ViewModel (ProblemStatementCellViewModel).

File: Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift (hunk 2 above)
Comment (huber-florian) [reply Themis/ViewModels/CorrectionSidebar/ProblemStatementCellViewModel.swift:115]: It is in a ViewModel (ProblemStatementCellViewModel).

**Pull request merged**